src/TwoBucks/UpdateProfile.java

```diff
@@ -1,75 +1,96 @@
-                dataRows[numOfRows++] = rowItems;
+        while((row=reader.readLine()) != null){ // understanding each value separated by comma
+            if(!line.strip().equals("")){ // check for empty string
+                rowItems = line.split(","); // separating values
+                dataRows[numOfRows++] = rowItems; //adding values to an array
             }
         }
 
         Scanner scnr = new Scanner(System.in);
         int option;
-        System.out.println("What would you like to update?");
+        System.out.println("What would you like to update?"); // menu
         System.out.println("");
         System.out.println("1. First Name");
         System.out.println("2. Last Name");
         System.out.println("3. Email");
-        System.out.println("");
-        System.out.println("Option: ");
 
         option = scnr.nextInt();
 
         if (option == 1){
 
-            int rowIndex = 0;
+            int rowIndex = 0; // set the index to 0 to be the position of first name
 
             System.out.println("Please confirm your existing first name: ");
-            String oldFirstName = scnr.next();
+            String oldFirstName = input.next();
             System.out.println("Please enter your new first name: ");
-            String newFirstName = scnr.next();
+            String newFirstName = input.next();
 
             Boolean flag = false;
 
-            for (i=0; i<numOfRows; i++){
+            for (i=0; i<numOfRows; i++){ // loop to check for name
 
                 if(dataRows[i][rowIndex].strip().equals(oldFirstName.strip())){
 
-                    dataRows[i][rowIndex] = newFirstName;
-                    flag = true;
+                    dataRows[i][rowIndex] = newFirstName; // insert the new name
+                    flag = true; // name is found
                     break;
                 }
             }
 
             if (flag == true){
-                BufferedWriter buffWriter = new BufferedWriter(new FileWriter(txtFile));
+                BufferedWriter buffWriter = new BufferedWriter(new Filewriter(txtFile));
 
                 for (i=0; i<numOfRows; i++){
-                    buffWriter.write(String.join(",",dataRows[i]) + "\n");
+                    buffWriter.write(String.join(",",dataRows[]) + "\n"); // new entry back into the file separating values with comma
                 }
                 buffWriter.close();
             }
             else{
-                System.out.println("Name not found.");
+                System.out.println("Name not found.")
             }
 
 
         }else if(option == 2){
             int rowIndex = 1;
 
             System.out.println("Please confirm your existing last name: ");
-            String oldLastName = scnr.next();
+            String oldLastName = input.next();
             System.out.println("Please enter your new last name: ");
-            String newLastName = scnr.next();
+            String newLastName = input.next();
```

Comment:
consider using setLastName method from the User class

Reply:
Done